### ISSUE TITLE
Missing env vars in build script

### ISSUE THREAD

**Issue by FHTMitchell** [prefix-dev/rattler-build#889]
Many env vars are missing in the build scripts (`$PY_VER, $STDLIB_DIR, $SP_DIR`) if no `dependencies.host` section is defined but `python {{ python }}` **is** in `dependencies.build`. The [docs](https://prefix-dev.github.io/rattler-build/latest/recipe_file/#host) state that if there is no `depencies.hosts`, then it is assumed to be the same as `dependencies.build` but here it is not the case. You can see this in the source [here](https://github.com/prefix-dev/rattler-build/blob/main/src/env_vars.rs#L47-L55).

**Comment by wolfv** [comment]
Indeed. I believe conda-build has this behavior for backwards compatibility - something we don't have to be so concerned about. It does simplify things in a few places to only use the host env and remove a corner case. Do you find it very painful? Would love to hear your experiences.

**Comment by matthewfeickert** [comment]
@wolfv so it seems that for [build variables](https://rattler.build/v0.31.1/build_script/#default-environment-variables-set-during-the-build-process), though `SHLIB_EXT` is available in the `tests` section, `PY_VER` is not. So if this is needed for doing file structure tests (c.f. https://github.com/conda-forge/staged-recipes/pull/29440/files#r2002297833) then is the recommendation to manually set the version yourself and export it to a local environment variable, like

```yaml
- export PY_VER=$(python -c "import sys; print(f'{sys.version_info.major}.{sys.version_info.minor}')")
```

(which to be clear is easy enough to do, but just wanted to check I'm not missing something more general)?

**Comment by jaimergp** [comment]
Got bitten by this now in https://github.com/conda-forge/conda-standalone-feedstock/pull/96. `$PYTHON` is defined but `$SP_DIR` is not when only `build` is present. That was a bit surprising.

**Comment by jaimergp** [comment]
Hm, actually `$PYTHON` is defined, but wrongly as `$PREFIX/bin/python`, which does not exist in `host`-less builds.
